### PR TITLE
Fix: Full width sidebar doesn't have a close button (#3802)

### LINKED ISSUE #3802: Full width sidebar doesn't have a close button
### Overview of the problem

**Buefy** version: [0.9.22]
**Vuejs** version: [3.2.40]
**OS/Browser**: Windows 11/Brave

### Description

Full width sidebar does not have a close button. It can be closed by pressing the `Esc` key however. This behaviour can confuse a user who is desperately trying to close the sidebar but is unable to find the close/exit button.

**Note:** As a controlled component, one can always add a button inside the sidebar to hide it. That is not the point of this bug report.

### Steps to reproduce

1. Navigate to `Sidebar` component in the website
2. Turn ON the `fullwidth` switch
3. Click the `Show` button
4. A user won't be able to close unless s/he hits the `Esc` key

### Expected behavior

Provide an internal default button and a default internal location for that `Close` button for the end-user to be able to close the `fullwidth`ed Sidebar

### PR DESCRIPTION
**Pull request by amal-qb**
Fixes #3802 

## Proposed Changes

- Fixes #3802

The close button is similar to Modal's close button.

image

### CODE REVIEW

**Comment by amal-qb** [comment]
@wesdevpro Can you please review this PR?

**Comment by wesdevpro** [comment]
> @wesdevpro Can you please review this PR?

I can do that 😅

File: src/components/sidebar/Sidebar.vue [hunk 1]
```diff
@@ -243,6 +249,14 @@ export default {
             this.hasLeaved = true
             this.timer = null
             this.isDelayOver = false
+        },
+        /**
+         * Close sidebar if close button is clicked.
+         */
+        clickedCloseButton() {
+            if (this.isFixed) {
+                if (this.isOpen && this.fullwidth) this.cancel('outside')
```
Comment (wesdevpro) [inline src/components/sidebar/Sidebar.vue:258]: Please put `this.cancel('outside')` on a new line underneath your last IF statement

File: src/scss/components/_sidebar.scss
Comment (wesdevpro) [inline src/scss/components/_sidebar.scss:1]: Looks good 🎉

**Review by wesdevpro — CHANGES_REQUESTED** [review]

**Review by wesdevpro — APPROVED** [review]

**Comment by wesdevpro** [comment]
@amal-qb can you also add documentation for this?

**Comment by amal-qb** [comment]
@wesdevpro I have added documentation.

**Pull request merged**